scripts: add benchmark helper

tbg opened the pull request:
Inspired by #17218. In that case, the comment would have been to run

```
BENCHES=BenchmarkScan/Cockroach/count=10000/limit=0-24 \
PKG=./pkg/sql ./scripts/bench master collerr
```

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17223)
<!-- Reviewable:end -->

benesch commented on scripts/bench:
```diff
@@ -0,0 +1,21 @@
+
+Requires golang.org/x/perf/cmd/benchstat.
+EOF
+exit 1
```
Indent?

tschottdorf replied:
Done.

benesch commented on scripts/bench:
```diff
@@ -0,0 +1,21 @@
+
+for branch in "${2}" "${1}"; do
+  git checkout "${branch}"
+  make bench PKG=${PKG} BENCHES=${BENCHES} TESTFLAGS="-count 10" > "${dest}/bench.${branch}" 2> "${dest}/log.txt"
```
Shellcheck would want you to quote PKG and BENCHES here too.

tschottdorf replied:
Done. Shellcheck is now happy.

benesch commented on scripts/bench:
```diff
@@ -0,0 +1,21 @@
+set -euo pipefail
+
+if [ $# -ne 2 ]; then
+  cat <<EOF
```
Stderr? cat >2 <<EOF

tschottdorf replied:
Done, though it had to be `cat >2 <<EOF ..... EOF\n >2` or nothing would come out.

tschottdorf replied:
Sorry, `cat 1>&2 <<EOF ...`

benesch reviewed (approved):
LGTM

tbg committed:
scripts: add benchmark helper

Inspired by #17218. In that case, the comment would have been to run

```
BENCHES=BenchmarkScan/Cockroach/count=10000/limit=0-24 \
PKG=./pkg/sql ./scripts/bench master collerr
```

benesch commented:
👍 sorry for the `>&2` typo
